Question: What is the planed strategy for using Noobaa-Function with additional modules needed.

Is there any documentation available on how to include additional modules used by functions?

We tried adding modules within a zip-file with a subfolder (./node_modules/…) which resulted in a function that can no longer be triggered (Errorcode: “new invocation failed”).

If we build core-images on our own we include modules but lack the possibility of automatic updates.

@jeniawhite 
Can you help us with this issue, it would help us a lot to add modules on demand. 
If we try to add modules in a subfolder ``` packages``` and get following error:
```
May-8 11:32:00.907 [WebServer/41] [ERROR] CONSOLE:: _load_func_code: FAILED Error: EEXIST: file already exists, mkdir '/tmp/functions_loading/k9y4dhq6/package'
--
May-8 11:32:00.912 [WebServer/41] [ERROR] CONSOLE:: RPC._on_request: ERROR srv func_api.invoke_func reqid 893@wss://noobaa-mgmt-noobaa-test.some-path.com(11idd0n.zu8r) connid ws://[::ffff:10.129.2.1]:53186/(8rmeekve) Error: EEXIST: file already exists, mkdir '/tmp/functions_loading/k9y4dhq6/package'

```

--- a/src/util/zip_utils.js
+++ b/src/util/zip_utils.js
@@ -91,9 +91,10 @@ function unzip_to_mem(zipfile) {
 
 function unzip_to_dir(zipfile, dir) {
     return unzip_to_callback(zipfile, (entry, stream) => {
-        const path_name = path.resolve(dir, '.' + path.sep + entry.fileName.toString());
+        const filename = entry.fileName.toString();
+        const path_name = path.resolve(dir, '.' + path.sep + filename);
         // directory ends with '/'
-        if (path_name.endsWith('/')) {
+        if (filename.endsWith('/')) {
             return fs_utils.create_path(path_name);
         }
         return P.resolve()

